### DISCUSSION TITLE
ClassPathResourceProvider vfs

### DISCUSSION THREAD

**Discussion by carlossm-00** [jobrunr/jobrunr#675] [Q&A] [answered]
Hi,
I am working with jobrunr 5.3.3, and so far with oracledb to persist the jobs.
Now I am trying to change the database to redis or elasticsearch, so, to test it in local I have created a docker in windows with the image "redis:latest" but I am getting this error:

`Caused by: java.lang.IllegalArgumentException: Unknown FileSystem required vfs.`

The code I am using to use redis is:

```
    public StorageProvider storageProvider(JobMapper jobMapper) {
        final LettuceRedisStorageProvider lettuceRedisStorageProvider = new LettuceRedisStorageProvider(getRedisClient(),rateLimit().withoutLimits());
        lettuceRedisStorageProvider.setJobMapper(jobMapper);
        return lettuceRedisStorageProvider;
    }

    private RedisClient getRedisClient(){
        return RedisClient.create(RedisURI.create("redis",6379));
    }

```
This error is coming from here:
```
    private FileSystemProvider getFileSystemProviderByScheme(String scheme) {
        switch (scheme) {
            case "jar": return new JarFileSystemProvider();
            case "resource": return new ResourcesFileSystemProvider();
            case "file": return new PathFileSystemProvider();
            default: throw new IllegalArgumentException("Unknown FileSystem required " + scheme);
        }
    }
```
So, what does it means that I am using vfs instead of jar, resource or file? can I change that? I am not using the Spring framework, I am using jboss in the project.

Thanks in advance!

**Comment by rdehuyss** [comment, marked as answer]
Please see https://github.com/jobrunr/jobrunr/issues?q=is%3Aissue+vfs
